### PR TITLE
[Android] Update gradle version and other changes in android apps, CI modification to auto-build Android apps and upload artifacts

### PR DESCRIPTION
**Pull request by argrento**
Update gradle configs to remove errors described in https://discuss.tvm.apache.org/t/gradle-build-fail-for-android-rpc-demo/10493

Modified:
* https://github.com/apache/tvm/tree/main/apps/android_camera
* https://github.com/apache/tvm/tree/main/apps/android_deploy
* https://github.com/apache/tvm/tree/main/apps/android_rpc

Also add CI jobs to build apps and keep artifacts.

Previous PR: https://github.com/apache/tvm/pull/10867

### CODE REVIEW

**Comment by argrento** [comment]
CC @TejashShah @Lyken17

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -38,59 +38,125 @@ jobs:
...
+          name: android_deploy-debug.apk
+          path: ./apps/android_deploy/app/build/outputs/apk/debug/app-debug.apk
+      - name: Build android_camera
+        working-directory: apps/android_camera
+        run: |
+          mkdir -p app/src/main/assets/models/
+          export TVM_NDK_CC=${ANDROID_HOME}/ndk/24.0.8215888/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android30-clang++
+          export TVM_HOME=~/work/tvm/tvm
+          export PYTHONPATH=$TVM_HOME/python:${PYTHONPATH}
+          pip3 install numpy decorator attrs scipy keras mxnet tensorflow
```
[the remark this answers is not in the document]
Comment (tmoreau89) [reply .github/workflows/main.yml:152]: good suggestion @argrento is that something we can add here?

**Review by tmoreau89 — COMMENTED** [review]
Thank you @argrento for keeping the android RPC app up to date and functional. Let @KJlaccHoeUM9l if the feedback can be addressed!

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -38,59 +38,125 @@ jobs:
...
+          ../tests/scripts/task_config_build_jvm.sh .
+          cmake ..
+          make -j8
+      - name: Build TVM4J
+        run: |
+          make jvmpkg
+      - name: Build android_rpc
+        working-directory: apps/android_rpc
+        run: |
+          export PATH="${ANDROID_HOME}/ndk/24.0.8215888:$PATH"
```
[the remark this answers is not in the document]
Comment (argrento) [reply .github/workflows/main.yml:128]: Not sure, if something like this is available.

[the remark this answers is not in the document]
Comment (argrento) [reply .github/workflows/main.yml:128]: It is available, I was wrong. 
https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md#android

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -38,59 +38,124 @@ jobs:
...
+          path: ./apps/android_deploy/app/build/outputs/apk/debug/app-debug.apk
+      - name: Build android_camera
+        working-directory: apps/android_camera
+        run: |
+          mkdir -p app/src/main/assets/models/
+          export TVM_NDK_CC=${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android30-clang++
+          export TVM_HOME=~/work/tvm/tvm
+          export PYTHONPATH=$TVM_HOME/python:${PYTHONPATH}
+          python3 ../../python/gen_requirements.py
+          pip3 install -r ../../python/requirements/core.txt
```
Comment (KJlaccHoeUM9l) [inline .github/workflows/main.yml:150]: It is better to replace relative paths through the environment variable `TVM_HOME`.

Comment (argrento) [reply .github/workflows/main.yml]: Done

**Comment by tmoreau89** [comment]
@mehrdadh and @jwfromm wanna take a look?

**Comment by Lyken17** [comment]
Confirmed can be built on TVM 0.8 and latest commit https://github.com/apache/tvm/commit/d1467777c66d20f32008c9cbf55d64f0cd8e401a on Mac OSX(OpenJDK 1.8.0_312 + Clang 13) and Ubuntu(OpenJDK 1.8.0 + GCC 9).


The CI scripts LGTM, thanks for @argrento effort! Android demo and rpc are important features!

**Pull request merged**